Init project

fickryiman opened the pull request:
Initialize the project with components:
- remove create react app boilerplate files
- add image, purposing to split the bookstore application into each component
- created BookstoreApp component
- created BookstoreInput component
- created BookstoreList component
- created BookstoreLogic component
- created BookstoreItem component
- created Header component
- add react router

fickryiman committed:
add more information to readme file

- add image, purposing to split the bookstore application to each components

fickryiman committed:
install react router

fickryiman committed:
initialize project with components

- created BookstoreApp component
- created BookstoreInput component
- created BookstoreList component
- created BookstoreLogic component
- created BookstoreItem component
- created Header component
- remove create react app boilerplate files

fickryiman committed:
linter fixes

fickryiman committed:
remove tested eslintrc file

amon-cofie commented on README.md:
You have a very detailed README.md file 🔥 But it seems that the demo link in the ```live demo``` section (Line 190) leads to a different project (math magician). Please update this link to lead to a deployed version of this project or you can remove the section if you don't have not deployed yet 👍

fickryiman replied:
Removed live demo URL.

amon-cofie commented on src/index.js:
```diff
@@ -1,17 +1,12 @@
   <React.StrictMode>
-    <App />
+    <BookstoreApp />
   </React.StrictMode>,
```
You have done well creating the index.js file and making it serve as the entry point for your app 👍. However, I suggest that you also create an App.js file in the ```src``` directory and import that in your index.js file so that you can manage all the functions and components of the app in the App.js file and reserve the index.js file for more system kernel related operations and configurations. This is not only conventional but also helps to add a more definitive and clean file structure to your project 😉.

fickryiman replied:
replaced components <BookstoreApp /> to <App />

amon-cofie commented on src/components/BookstoreApp.jsx:
```diff
@@ -0,0 +1,11 @@
+  </>
+);
+
+export default BookstoreApp;
```
I can see from the package.json file that you installed the react-router-dom 👍. But I realise that you didn't use it in your app. I suggest that you take advantage of the ability of the react-router-dom to allow us to manage routing to different pages without reloading the page to create the two pages required in the project requirements (books and categories). 👍 

In the end your component should look something like this: 

```sh
import React from "react"
import { BrowserRouter, Routes, Route, Link } from "react-router-dom"

function App() {
  return (
    <BrowserRouter>
      <Navbar />
      <Routes>
        <Route path="/" element={<}>
          <Route index element={<Books />} />
          <Route path="/categories" element={<Categories />} />
        </Route>
      </Routes>
    </BrowserRouter>
  )
}

```

fickryiman replied:
add BrowserRouter, Routes, Route, Link from react-router-dom

amon-cofie commented on src/components/BookstoreInput.jsx:
```diff
@@ -0,0 +1,5 @@
+const BookstoreInput = () => (
+  <div>input field here...</div>
+);
+
+export default BookstoreInput;
```
I suggest that you use a ```form``` element with inputs for the title and the author of the book as well as a submit button instead of a ```div```. This is because the page would be requiring a user to make some inputs and using a form would be appropriate to handle those inputs. 👍

fickryiman replied:
added Form Input

amon-cofie reviewed (changes requested):
Hi @fickryiman , 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! 💪 

<img src="https://media0.giphy.com/media/VlqzeaRnhCvPLfrSh6/giphy.gif"/>

## Highlights

🚀 You have a descriptive pull request summary
🚀 You have good commit messages
🚀 You used Gitflow properly
🚀 You have no linter errors

## Required Changes ♻️

- [ ] I suggest that you tweak you file structure a little to match the requirements. Consider making your src directory look something like this: 
```
| src
   | components
      | books
         | Book.jsx
         | Bookslist.jsx
         | NewBookForm.jsx
      | categories
   | app.js
   | index.js
```

- [ ] Please remember to add the categories component even though it is supposed to be empty since it is part of the project requirements.

I am available on slack if you need any help and please do not hesitate to contact me.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

fickryiman committed:
changes components file name

- add App.js

fickryiman committed:
removed live demo URL and will provide live demo URL after finished the project

fickryiman committed:
linters fixes

fickryiman committed:
add styles to each component

fickryiman committed:
linters fixes

sumairq reviewed (approved):
### Hi @fickryiman , 

<img src="https://media4.giphy.com/media/BPJmthQ3YRwD6QqcVD/giphy.gif"/>

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

fickryiman committed:
add profile image

fickryiman committed:
changes and UI

- add react icon (npm install react-icons)
- add styles
- make sure no linters error
